error.c

```diff
@@ -77,6 +77,7 @@ static ID id_deprecated;
 static ID id_experimental;
 static VALUE sym_category;
 static VALUE warning_categories;
+static VALUE warning_category_t_map;
```

Comment:
If it was performance sensitive, that's probably a better approach, however, since this is related to emitting warnings, I think a Ruby hash is simpler.  I do agree to switch to an ident hash is better.